pkg/server/plugin/nodeattestor/gcp/iit.go

```diff
@@ -17,74 +19,80 @@ import (
 const (
 	tokenAudience = "spire-gcp-node-attestor"
 	googleCertURL = "https://www.googleapis.com/oauth2/v1/certs"
+	svidPrefix    = "spiffe://{{ .TrustDomain }}/spire/agent"
 )
 
+var defaultAgentSVIDTemplate = template.Must(template.New("agent-svid").Parse(fmt.Sprintf("%s/{{ .PluginName}}/{{ .ProjectID }}/{{ .InstanceID }}", svidPrefix)))
+
 type tokenKeyRetriever interface {
 	retrieveKey(token *jwt.Token) (interface{}, error)
 }
 
+// IITAttestorPlugin implements node attestation for agents running in GCP.
+type IITAttestorPlugin struct {
+	svidTemplate      *template.Template
+	config            *IITAttestorConfig
+	mtx               sync.Mutex
+	tokenKeyRetriever tokenKeyRetriever
+}
+
+// IITAttestorConfig is the config for IITAttestorPlugin.
 type IITAttestorConfig struct {
 	trustDomain        string
 	ProjectIDWhitelist []string `hcl:"projectid_whitelist"`
+	AgentSVIDTemplate  string   `hcl:"agent_svid_template"`
 }
 
-type IITAttestorPlugin struct {
-	tokenKeyRetriever tokenKeyRetriever
+// NewIITAttestorPlugin creates a new IITAttestorPlugin.
+func NewIITAttestorPlugin() *IITAttestorPlugin {
+	return &IITAttestorPlugin{
+		svidTemplate:      defaultAgentSVIDTemplate,
+		tokenKeyRetriever: newGooglePublicKeyRetriever(googleCertURL),
+	}
+}
 
-	mtx    sync.Mutex
-	config *IITAttestorConfig
+// templateData is the data passed to the agent SVID template.
+type templateData struct {
```

Comment:
nit: i have a feeling this struct and the code that does the template execution will be needed in the agent PR as well. I'd be fine with this staying as is in this PR but ideally it would make it's way into a common package (e.g. `pkg/common/plugin/gcp`) for consumption by the agent side code.